src/chrome/chromeUtils.ts

```diff
@@ -312,3 +315,45 @@ export function getUrlRegexForBreakOnLoad(url: string): string {
     const escapedFileName = pathToRegex(fileNameWithoutExtension);
     return '.*[\\\\\\/]' + escapedFileName + '([^A-z^0-9].*)?$';
 }
+
+/**
+ * Checks if a given tcp port is currently in use (more accurately, is there a server socket accepting connections on that port)
+ * @param port The port to check
+ * @param host Optional host, defaults to 127.0.0.1
+ * @param timeout Timeout for the socket connect attempt
+ * @returns True if a server socket is listening on the given port, false otherwise
+ */
+export async function isPortInUse(port: number, host: string = '127.0.0.1', timeout = 400): Promise<boolean> {
+    // Basically just create a socket and try to connect on that port, if we can connect, it's open
+    return new Promise<boolean>((accept, _reject) => {
+        const socket = new Socket()
+        const callback = (inUse: boolean) => () => {
+            accept(inUse);
+            socket.removeAllListeners();
+            socket.destroy();
+        }
+
+        socket.setTimeout(timeout)
+        socket.on('connect', callback(true));
+        socket.on('timeout', callback(false));
+        socket.on('error', callback(false));
+        socket.connect(port, host);
+    });
+}
+
+/**
+ * Get the port on which chrome was launched when passed "--remote-debugging-port=0"
+ * @param userDataDir The profile data directory for the Chrome instance to check
+ * @throws If reading the port failed for any reason
+ */
+export async function getLaunchedPort(userDataDir: string): Promise<number> {
+    let port = 0;
+    const activePortFilePath = path.join(userDataDir as any, 'DevToolsActivePort');
+    const activePortArgs = (await promisify(fs.readFile)(activePortFilePath)).toString();
```

Comment:
There is some readFileP around that does this same thing

Reply:
Changed